libs/utils/utils.js

```diff
@@ -1845,16 +2045,45 @@ export async function loadArea(area = document) {
   }
 
   if (isDoc) {
-    decorateDocumentExtras();
+    await decorateDocumentExtras();
+    initModalEventListener();
+  }
+
+  if (lingoActive()) {
+    const hasRegions = Object.keys(config.locale?.regions || {}).length > 0;
+    if (hasRegions) {
+      const country = new URLSearchParams(window.location.search).get('akamaiLocale')?.toLowerCase()
+        || sessionStorage.getItem('akamai');
+      const prefixParts = config.locale.prefix.split('/').filter((part) => part);
+      const localeCode = prefixParts[0];
+      const regionKey = `${country}_${localeCode}`;
+      const userIsInRegion = country && config.locale.regions[regionKey];
+      if (userIsInRegion) {
+        const targetPrefix = userIsInRegion
+          ? config.locale.regions[regionKey].prefix
+          : config.locale.prefix;
+        loadQueryIndexes(targetPrefix);
+      }
+    } else if (config.locale?.base) {
+      const { prefix } = config.locale;
+      loadQueryIndexes(prefix);
+    }
   }
 
-  const sections = decorateSections(area, isDoc);
+  const htmlSections = [...area.querySelectorAll(isDoc ? 'body > main > div' : ':scope > div')];
+  htmlSections.forEach((section) => { section.className = 'section'; section.dataset.status = 'pending'; });
+
+  // Preload mep-lingo module if lingo is active (fire and forget for LCP performance)
+  if (lingoActive()) {
```

Comment:
nit: We could call `lingoActive()` once towards the top of the file and then use a const instead I think